andela/travel_tool_front
# #160157497 Setup bugsnag for error reporting

#### What does this PR do?
- It integrates bugsnag with the application as a tool for error reporting.

#### Description of Task to be completed?
If there are application errors while users interact with the app, the app should be able to track these errors. Tools that can be used for this include;  Kibana, Bugsnag, Stackdriver.

Acceptance criteria
All automatic failures from both the backend and frontend should be sent and visible from the bugsnag dashboard.

#### How should this be manually tested?
1. Create an account at [bugsnag](https://www.bugsnag.com/).
2. Create a project, and copy it's API key.
3. Add the API key in the .env file:
`REACT_APP_BUGSNAG_API_KEY=your-bugsnag-api-key`
4. Break something in the code, and run the application.
5. Refresh the dashboard page in bugsnag, and you shall see the error.

#### Any background context you want to provide?
If the `REACT_APP_BUGSNAG_API_KEY` is not provided, then the application would work normally without reporting errors to bugsnag.

#### What are the relevant pivotal tracker stories?
#160157497

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-09-06 at 12 34 31" src="https://user-images.githubusercontent.com/18000724/45149126-860bc680-b1d1-11e8-9e0b-4878fd1de0ac.png">
